src/start_server.ts

```diff
@@ -81,44 +80,193 @@ async function applyDefaultOptions(options: ServerOptions):
     Promise<http.Server> {
-      options = options || {};
-      assertNodeVersion(options);
-      try {
-        return await startWithPort(options)
-      } catch (e) {
-        console.error('ERROR: Server failed to start:', e);
-        throw new Error(e);
-      }
+      return (await _startServer(options)).server;
+    }
+
+async function _startServer(options: ServerOptions) {
+  options = options || {};
+  assertNodeVersion(options);
+  try {
+    const fullOptions = await applyDefaultOptions(options);
+    const app = getApp(options);
+    const server = await startWithPort(fullOptions, app);
+    return {app, server};
+  } catch (e) {
+    console.error('ERROR: Server failed to start:', e);
+    throw new Error(e);
+  }
+}
+
+export type ServerInfo = MainlineServer | VariantServer | ControlServer;
+
+/**
+ * The `default` or `primary` server. If only one ServerInfo is returned from
+ * startServers it must be a MainlineServer. This is the server that's running
+ * with the default configuration and not running a variant configuration.
+ */
+export interface MainlineServer {
+  kind: 'mainline';
+  server: http.Server;
+  app: express.Application;
+  options: ServerOptions;
+}
+/**
+ * These are servers which are running some named variant configuration. For
+ * multiple variant dependency directories are detected/configured, there will
+ * be one MainlineServer that serves out the default dependency directory, and
+ * one VariantServer for each other dependency directory.
+ */
+export interface VariantServer {
+  kind: 'variant';
+  server: http.Server;
+  app: express.Application;
+  options: ServerOptions;
+  variantName: string;
+  dependencyDir: string;
+}
+/**
+ * If more than one server is started by startServers, the main port will serve
+ * out a control server. This server serves out an HTML interface that
+ * describes the other servers which have been started, and provides convenience
+ * links to them.
+ */
+export interface ControlServer {
+  kind: 'control';
+  server: http.Server;
+  app: express.Application;
+  options: ServerOptions;
+}
+
+export interface MultipleServersInfo {
+  kind: 'MultipleServers';
+  mainline: MainlineServer;
+  variants: VariantServer[];
+  control: ControlServer;
+}
+
+export type StartServerResult = MainlineServer | MultipleServersInfo;
+
+/**
+ * Starts one or more web servers, based on the given options and
+ * variant bower_components directories that are found in the root dir.
+ */
+export async function startServers(options: ServerOptions):
+    Promise<StartServerResult> {
+  const variants = await findVariants(options);
+  // TODO(rictic): support manually configuring variants? tracking more
+  //   metadata about them besides their names?
+  if (variants.length > 0) {
+    return await startVariants(options, variants);
+  }
+
+  const serverAndApp = await _startServer(options);
+  return {
+    kind: 'mainline',
+    server: serverAndApp.server,
+    app: serverAndApp.app, options
```

Comment:
Ah, this repo doesn't have a tslint. I'll add that in a follow up PR.

Moved options up so that it's on its own line